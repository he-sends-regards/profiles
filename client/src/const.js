export const AppRoute = {
  ROOT: '/',
  AUTH: '/auth',
  ACCOUNT: '/account',
};

export const APIRoute = {
  AUTH: '/api/auth',
  GET_USERS: '/api/users',
  DASHBOARD: '/api/users/dashboard',
  DELETE_USER: '/api/users/delete',
  GET_PROFILES: '/api/profiles/',
};

export const AuthorizationStatus = {
  AUTH: 'AUTH',
  NO_AUTH: 'NO_AUTH',
};

export const HTTPStatus = {
  OK: 200,
};

export const ProfileFormType = {
  CREATE: 'CREATE',
  EDIT: 'EDIT',
};
