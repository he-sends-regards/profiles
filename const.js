const PORT = 5000;
const jwtSecret = 'profiles1testtask1secret1string1';
const mongoUri = 'mongodb+srv://postscriptumno:HelloFrom2021@cluster0.zlo8z.mongodb.net/db?retryWrites=true&w=majority';

module.exports = {PORT, jwtSecret, mongoUri};
