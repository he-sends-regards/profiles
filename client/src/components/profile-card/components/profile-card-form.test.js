import React from 'react';
import renderer from 'react-test-renderer';
import {ProfileFormType, MenuItem} from '../../../const.js';
import ProfileCardForm from './profile-card-form.jsx';

const profile = {
  _id: '1',
  owner: 'hello@gmail.com',
  birthdate: '2021-02-13',
  gender: 'Male',
  name: 'Harry Potter',
  city: 'London',
};

const noop = () => {};

it('Renders correctly', () => {
  const tree = renderer
      .create(
          <ProfileCardForm
            profile={profile}
            type={ProfileFormType.CREATE}
            listType={MenuItem.PROFILES_NETWORK.id}
            setIsProfileDataChanged={noop}
            setIsCardEditing={noop}
            setIsCardCreating={noop}
          />,
      )
      .toJSON();
  expect(tree).toMatchSnapshot();
});
