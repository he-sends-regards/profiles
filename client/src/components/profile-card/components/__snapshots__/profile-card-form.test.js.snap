// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<form
  className="user-card-edit"
  onSubmit={[Function]}
>
  <div
    className="user-card-edit__close-container"
  >
    <button
      className="user-card-edit__close_btn"
      onClick={[Function]}
    >
      <img
        alt="Close creation of the new profile"
        className="user-card-edit__close-img"
        src="cancel.svg"
      />
    </button>
  </div>
  <div>
    <label
      htmlFor="name-field"
    >
      Name:
    </label>
    <input
      defaultValue="Harry Potter"
      id="name-field"
      minLength="2"
      required={true}
      type="name"
    />
  </div>
  <br />
  <div>
    <label
      htmlFor="gender"
    >
      Gender:
    </label>
    <select
      defaultValue="Male"
      id="gender"
      name="gender"
      required={true}
    >
      <option
        value="Male"
      >
        Male
      </option>
      <option
        value="Female"
      >
        Female
      </option>
      <option
        value="Other"
      >
        Other
      </option>
    </select>
  </div>
  <br />
  <div>
    <label
      htmlFor="email-field"
    >
      Birthdate: 
    </label>
    <input
      defaultValue="2021-02-13"
      name="calendar"
      required={true}
      type="date"
    />
  </div>
  <br />
  <div>
    <label
      htmlFor="city-field"
    >
      City: 
    </label>
    <input
      defaultValue="London"
      id="city-field"
      required={true}
      type="text"
    />
    <br />
  </div>
  <div
    className="submit-btn"
  >
    <button
      className="btn btn-success"
      disabled={false}
      type="submit"
    >
      Create
    </button>
  </div>
</form>
`;
